File: pinyougou_parent/pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;

import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;

import com.pinyougou.pojo.TbBrand;

@RestController//@Controller 和@ResponseBody的结合,这样配置,相当于返回值直接回返回到请求的页面
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;


    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }


    //分页查询
    @RequestMapping("/findPage")
    public PageResult findPage(@RequestParam(name = "page", defaultValue = "1") Integer currentPage, @RequestParam(name = "size", defaultValue = "10") Integer pageSize) {

        return brandService.findPage(currentPage, pageSize);
    }


    //添加
    @RequestMapping("/saveBrand")
    public Result saveBrand(@RequestBody TbBrand brand) {
        try {
            brandService.saveBround(brand);
            return new Result(true, "添加成功");
        } catch (Exception e) {
            return new Result(false, "添加失败");
        }
    }

    //修改
    @RequestMapping("/updateBrand")
    public Result updateBrand(@RequestBody TbBrand brand) {
        try {
            brandService.changeBrand(brand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            return new Result(false, "修改失败");
        }
    }

    //删除
    @RequestMapping("/deleteBrand")
    public Result deleteBrand(Long[] ids) {
        try {
            brandService.deleteBrand(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            return new Result(false, "删除失败");
        }

    }


    //模糊查询
    @RequestMapping("/searchPage")
    public PageResult searchPage(@RequestParam(name = "page", defaultValue = "1") Integer currentPage, @RequestParam(name = "size", defaultValue = "10") Integer pageSize, @RequestBody TbBrand brand) {

        PageResult page = brandService.findPage(currentPage, pageSize, brand);
        return page;
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){

        List<Map> maps = brandService.selectOptionList();

        return maps;
    }
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/controller/brandController.js
app.controller("brandController", function ($scope, $http, brandService,$controller) {

    //引入父类的js文件
    $controller('baseController',{$scope:$scope});

    //查询所有
    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {
                $scope.pageResult = response;
            }
        )
    };



    //刷新页面
    $scope.reloadPage = function () {
        $scope.searchBrand($scope.paginationConf.currentPage, $scope.paginationConf.itemsPerPage)
    };


    //分页
    $scope.findPage = function (page, size) {
        brandService.findPage(page, size).success(
            function (response) {
                $scope.list = response.rows;//显示当前的数据
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        )
    };

    //保存品牌
    $scope.saveBrand = function () {

        if ($scope.entity.id == null) {
            //若果id值是null,说明是想数据库中添加数据
            brandService.saveBrand($scope.entity).success(
                function (response) {
                    $scope.Result = response;
                    if ($scope.Result.flag) {
                        $scope.reloadPage();
                    } else {
                        alert($scope.Result.msg);
                    }

                }
            )
        } else {
            //如果id不是null,说明是想数据库中修改数据
            brandService.updateBrand($scope.entity).success(
                function (response) {
                    $scope.Result = response;
                    if ($scope.Result.flag) {
                        $scope.reloadPage();
                    } else {
                        alert($scope.Result.msg);
                    }
                }
            )

        }


    };

    //修改品牌
    $scope.changeBrand = function (brand) {
        //把原来的数据赋值到表格中,并发送ajax请求
        $scope.entity = brand;
    };

    //删除品牌:

    //3调用删除方法,把ids传入到后台
    $scope.deleteBrand = function () {
        brandService.deleteBrand($scope.ids).success(
            function (response) {
                $scope.Result = response;
                if ($scope.Result.flag) {
                    $scope.reloadPage();
                } else {
                    alert($scope.Result.msg);
                }
            }
        )
    };


    $scope.searchEntity = {};//初始化searchEntity,保证接收的不为null;
    //条件查询:
    $scope.searchBrand = function (page, size) {
        brandService.searchBrand(page, size,$scope.searchEntity).success(
            function (response) {
                $scope.list = response.rows;//显示当前的数据
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        )
    };




});